### ISSUE TITLE
Non standard characters in filenames break updating logic

### ISSUE THREAD

**Issue by Vocllum** [Skidamek/AutoModpack#297]
### Bug description

The server can synchronize files normally, but whether it's synchronizing the required files or preparing the required files on the client side, when there are no files to update, the game will still be required to be restarted
I have tried the solutions for the previous issue, but it still hasn't been resolved
The module name contains Chinese characters, but the converted module displays gibberish. I suspect that the module's checking mechanism is causing the discrepancy between the original name and the converted name?
Chinese community players usually use the PCL launcher, and the mods downloaded on it (usually from CurseForge or ModRith mirror sources) may contain Chinese content. Unfortunately, this seems to be unavoidable.
So, regardless of whether it's due to this or not, could you make the module support Chinese text? I also hope there are other solutions

### Steps to reproduce

enter the server, ask for update, update, reboot, enter the server...

### Expected behavior

enter the server

### Actual behavior

need to reboot

### Minecraft & Mod Loader versions

1.20.1 Forge

### Minecraft launcher

PCL 2

### Operating system

win11

### AutoModpack version

4.0.0 beta18

### Check list

- [X] I have verified that the issue persists in the latest version of the mod.
- [X] I have searched the existing issues and confirmed that this is not a duplicate.

**Comment by Skidamek** [comment]
So you are suspecting that this issue is caused because of Chinese characters in the filename, is it right?

**Comment by Vocllum** [comment]
Yeah but Im not sure. Anyway, Im trying to find some way to work it out, too

**Comment by Skidamek** [comment]
Because #289 and #266 seem similar besides of Chinese characters, have you read them?

**Comment by Vocllum** [comment]
Yes, I have read them, but none of them can solve the problem. I just tried to keep only a few modules without Chinese content, and they ran well

**Comment by Skidamek** [comment]
To be honest no idea why it would do it, could you give an example filename which breaks downloads?

**Comment by Vocllum** [comment]
I don't know why the mod automatically upgraded from beta20 to beta18 and repeatedly requested updates and game restarts. I don't have time to correct and demonstrate it now, so I will try to describe the phenomenon to you in detail. All content is translated using AI, so please forgive any errors!
Suppose I have a mod named "农夫乐事 - FarmersDelight-1.20.1-1.2.5.jar"
After conversion, the file "   璘輲?-FarmersDelight-1.20.1-1.2.5.jar" appears in the host-modpack folder on the server (the gibberish is meaningless. I copied this text from elsewhere to show you that it is different from the original text, which is similar to common encoding errors, so I guess it is caused by encoding). However, the mod name is displayed correctly in automodpack-content.json.
Even if you install the complete mod pack on the client side, when you enter the server, it will display "No Updates Available", with a download progress of 0/0. Then it will show "Update Complete" and ask you to restart the game

**Comment by Skidamek** [comment]
So it probably downgrades you to beta 18 because on sever you got beta 18, update server to beta 20 and client should automatically update to match the same version.

> Even if you install the complete mod pack on the client side, when you enter the server, it will display "No Updates Available", with a download progress of 0/0. Then it will show "Update Complete" and ask you to restart the game

Could you send a client log from when that happens?

**Comment by Vocllum** [comment]
[debug1.zip](https://github.com/user-attachments/files/17897157/debug1.zip)
[debug2.zip](https://github.com/user-attachments/files/17897158/debug2.zip)
Today, there was no downgrade from beta20 to beta18. However, it's worth mentioning that the update occurred when the game was launched, so it seems to be unrelated to the server.
I installed the complete mod on the client side and copied it to the server host modpack, with all settings set to default.
The complete modlist comprises 199 mods, and approximately 290 mods are loaded upon startup (due to the inclusion of Sinytra Connector, which, if you're unaware, serves as a mod that acts as a translation/compatibility layer to enable Forge to run Fabric mods). Upon first entering the server, I was prompted to update my mods, resulting in the download of approximately 350 files. Please refer to the attached debug1 for details.
The second time I entered the server, it updated 0 files and I was required to restart the game. See debug2 for details. At this time, there was an error on the server side:
[AutoModpack/]: Error writing to channel: null path: C:\MC\VVMS\VestalValley\mods\.connector\better-end-4.0.11_mapped_srg_1.20.1.jar io.netty.channel. StacklessClosedChannelException
The .connector folder is generated by Sinytra Connector upon the first launch of the game, and I did not place this folder under the host modpack folder.
After set ".connector" folder still has no available file updates, and the game needs to be restarted

**Comment by Skidamek** [comment]
Looks like its also issue with chars like `§`
